pkg/ddl/tests/partition/db_partition_test.go

```diff
@@ -3658,3 +3658,22 @@ func checkGlobalAndPK(t *testing.T, tk *testkit.TestKit, name string, indexes in
 		require.True(t, idxInfo.Primary)
 	}
 }
+
+func TestAdminCheckWithPartitions(t *testing.T) {
```

Comment:
It similar with some `admin check` test cases in `pkg/executor/test/admintest/admin_test.go`. Maybe we can modify those cases directly instead of add a new one.